app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -27,18 +41,74 @@ public LoginPresenter(SharedPreferencesManager mSharedPreferencesManager) {
      * @param password Input password
      */
     void login(@NonNull String email, @NonNull String password) {
-        mSharedPreferencesManager.store(Constants.UserCredentials.USER_EMAIL, email);
-        mSharedPreferencesManager.store(Constants.UserCredentials.USER_PASSWORD, password);
-        runIfViewAttached(ILoginView::onUserSaved);
+        runIfViewAttached(ILoginView::showProgress);
+
+        mUserService.login(email, password).enqueue(new SimpleNetworkCallback<List<User>>() {
+            @Override
+            public void onResponseSuccessful(@Nullable List<User> users) {
+                runIfViewAttached(ILoginView::hideProgress);
+
+                if (users == null || users.size() == 0) {
+                    runIfViewAttached(ILoginView::onWrongEmailOrPassword);
+                    return;
+                }
+
+                User user = users.get(0);
+                LoginPresenter.this.storeEmail(user.getEmail());
+                mSharedPreferencesManager.store(Constants.UserCredentials.USER_LOGGED_IN, true);
+
+                runIfViewAttached(ILoginView::onUserLoggedIn);
+            }
+
+            @Override
+            public void onResponseFailed(@Nullable ResponseBody responseBody, int statusCode) {
+                super.onResponseFailed(responseBody, statusCode);
+
+                runIfViewAttached(iLoginView -> {
+                    iLoginView.hideProgress();
+                    iLoginView.showUnexpectedError();
+                });
+            }
+
+            @Override
+            public void onCallFailure(@Nullable Throwable exception) {
+                super.onCallFailure(exception);
+
+                runIfViewAttached(iLoginView -> {
+                    iLoginView.hideProgress();
+
+                    if (exception instanceof UnknownHostException) {
+                        iLoginView.showNoConnectionError();
+                    } else {
+                        iLoginView.showUnexpectedError();
+                    }
+                });
+            }
+        });
     }
 
     /**
      * Load saved email & password
      */
-    void loadEmailAndPassword() {
-        runIfViewAttached((view) -> {
-            view.setEmail(mSharedPreferencesManager.get(Constants.UserCredentials.USER_EMAIL, ""));
-            view.setPassword(mSharedPreferencesManager.get(Constants.UserCredentials.USER_PASSWORD, ""));
-        });
+    void loadEmail() {
+        runIfViewAttached((view) -> view.setEmail(mSharedPreferencesManager.get(Constants.UserCredentials.USER_EMAIL, "")));
```

Comment:
Please don't leave empty the second argument of get. Use for example a String called DEFAULT_USER, defined at the top of the class